src/Pages/MainPage.tsx

```diff
@@ -1,15 +1,108 @@
 import styled from '@emotion/styled';
 import { QuizListContent } from '../Components/QuizListContent';
 import { QuizListHeader } from '../Components/QuizListHeader';
+import { useNavigate } from 'react-router-dom';
+import { useParams,Link } from 'react-router-dom';
+import { useSelector } from 'react-redux';
+import { useCallback } from 'react';
+import { RootState } from '../state/store';
+import { useEffect, useState } from 'react';
+import axios from 'axios';
+import { qinfoType } from '../apiTypes/qinfo';
+import { clusterType } from '../apiTypes/cluster';
+
+interface propsType{
+	createOptions: boolean
```

Comment:
I think it is better to keep this state in Redux. Page components are the highest level components in App, so it seems not intuitive to me that they have props from parents. What do you think?

Reply:
> I think it is better to keep this state in Redux. Page components are the highest level components in App, so it seems not intuitive to me that they have props from parents. What do you think?

Yes, you are right I think an even better way could be just reading the URL and redux won't be necessary; I updated it this way now :)